core/controllers/suggestion_test.py

```diff
@@ -475,3 +592,140 @@ def test_accept_question_suggestion(self):
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                suggestion_models.SUGGESTION_TYPE_ADD_QUESTION)
+            )['suggestions'][0]
+
+        suggestion_id = suggestion_to_accept['suggestion_id']
+        invalid_suggestion_id = ('exploration.' +
+                                 '.'.join(suggestion_id.split('.')[1:]))
+
+        self.login(self.ADMIN_EMAIL)
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.put_json('%s/topic/%s/%s' % (
+                feconf.SUGGESTION_ACTION_URL_PREFIX,
+                suggestion_to_accept['target_id'],
+                invalid_suggestion_id), {
+                    'action': u'accept',
+                    'commit_message': u'commit message',
+                    'review_message': u'This looks good!',
+                    'skill_id': self.SKILL_ID
+                }, csrf_token=csrf_token, expected_status_int=400)
+
+    def test_accept_suggestion_on_invalid_exploration_id(self):
+        suggestion_to_accept = self.get_json(
+            '%s?suggestion_type=%s' % (
+                feconf.SUGGESTION_LIST_URL_PREFIX,
+                suggestion_models.SUGGESTION_TYPE_ADD_QUESTION)
+            )['suggestions'][0]
+
+        self.login(self.ADMIN_EMAIL)
+        response = self.get_html_response(feconf.CREATOR_DASHBOARD_URL)
+        csrf_token = self.get_csrf_token_from_response(response)
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_PLAYERS', True):
+            response = self.put_json('%s/topic/%s/%s' % (
+                feconf.SUGGESTION_ACTION_URL_PREFIX,
+                'random',
+                suggestion_to_accept['suggestion_id']), {
+                    'action': u'accept',
+                    'commit_message': u'commit message',
+                    'review_message': u'This looks good!',
+                    'skill_id': self.SKILL_ID
+                }, csrf_token=csrf_token, expected_status_int=400)
+            self.assertEqual(response['error'], 'The topic id provided does ' +
+                             'not match the topic id present as part of the ' +
+                             'suggestion_id')
+
+
+class SuggestionListHandlerTests(test_utils.GenericTestBase):
+
+    def test_suggestion_list_handler(self):
+        params = {
+            'target_type': 'exploration',
+            'target_id': 'exp_id'
+        }
+        response = self.get_json(
+            '/suggestionlisthandler', params=params)
+        expected_response = {'additional_angular_modules': [],
+                             'suggestions': [], 'is_super_admin': False,
+                             'is_moderator': False, 'iframed': False,
+                             'is_admin': False, 'is_topic_manager': False}
+        self.assertEqual(expected_response, response)
```

Comment:
Create a suggestion and again do a get_json. The suggestions field should be updated and test its new value